tests/tape/test_qscript.py

```diff
@@ -128,6 +130,21 @@ def test_cached_graph_specs_reset(self):
         assert qs._graph is None
         assert qs._specs is None
 
+    def test_update_measured_wires(self):
+        """Test that on construction measured_wires is set."""
+        prep = [qml.BasisState([1, 1], wires=(-1, -2))]
+        ops = [qml.S(0), qml.T("a"), qml.S(0)]
+        measurement = [
+            qml.probs(wires=(-1)),
+            qml.expval(qml.Hermitian(2 * np.eye(2), wires=-2)),
+            qml.PauliX(-1),
```

Comment:
While this is unvalidated during the initialization process,  `measurements` should be [`Iterable[MeasurementProcess]`](https://github.com/PennyLaneAI/pennylane/blob/597beac35f49874a042f6e0dc61bdff69cce19a3/pennylane/tape/qscript.py#L8). We don't perform this validation since `QuantumScript` is mostly internal and produced in controlled situations.

`PauliX` is an `Operator`, not a `MeasurementProcess`, so it shouldn't be in the `measurement` list.

Reply:
That's interesting. I am just getting to know Pennylane so thanks for bearing with me. Is it the case that I need to wrap it in qml.expval to get the expectation value of the observable?

Reply:
Exactly.  `qml.expval(qml.PauliX(-1))` indicates taking an expectation value measurement of the observable `PauliX`.